naga/src/front/spv/function.rs

```diff
@@ -376,34 +376,33 @@ impl<I: Iterator<Item = u32>> super::Frontend<I> {
                     // See the docs for `Frontend::gl_per_vertex_builtin_access`.
                     {
                         let ty = &module.types[result.ty];
-                        match ty.inner {
-                            crate::TypeInner::Struct {
-                                members: ref original_members,
-                                span,
-                            } if ty.name.as_deref() == Some("gl_PerVertex") => {
-                                let mut new_members = original_members.clone();
-                                for member in &mut new_members {
-                                    if let Some(crate::Binding::BuiltIn(built_in)) = member.binding
-                                    {
-                                        if !self.gl_per_vertex_builtin_access.contains(&built_in) {
-                                            member.binding = None
-                                        }
+                        if let crate::TypeInner::Struct {
+                            members: ref original_members,
+                            span,
+                        } = ty.inner
+                        {
+                            let mut new_members = None;
```

Comment:
I refactored this a bit so that it only clones the members list if it is going to be modified